### PR TITLE
Feature/859

### PR DESCRIPTION
**Pull request by fraincs**
Issue: 

#859 - Tag should have rounded corner (like a button)

## What I did

- Tags now have rounded corners.
- Tags have a revamped padding  when a dot / icon is present.
- Buttons now follow the same rules as Tags padding.
- Fixed an issue where the documentation Planet image could trick you into thinking that a component had too much padding.
- Fixed an issue where in certains situations a Switch control could go further than it's parent.

### CODE REVIEW

**Comment by patricklafrance** [comment]
Yyyyeeeesssss!!!

File: packages/components/src/button/src/Button.tsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ export function InnerButton(props: InnerButtonProps) {
         text: {
             "aria-hidden": loading,
             className: "o-ui-button-text",
-            size: condensed ? condensedTextSize(sizeValue) : sizeValue
+            size: condensed ? condensedTextSize(sizeValue) : "md"
```
Comment (patricklafrance) [inline packages/components/src/button/src/Button.tsx:147]: So you don't want the button text to accept a responsize size when it's not condensed?

File: packages/components/src/tag/src/Tag.tsx [hunk 2]
```diff
@@ -81,7 +81,7 @@ export function InnerTag({
         text: {
             className: "o-ui-tag-text",
             color: "inherit",
-            size: sizeValue
+            size: "md"
```
Comment (patricklafrance) [inline packages/components/src/tag/src/Tag.tsx:84]: So you don't want a tag text to accept a responsize size value?

**Comment by patricklafrance** [comment]
309 chromatic changes 😱 I hope you won’t miss anything!

Would be nice to find a way to avoid combining changes which results in to much changes in chromatic.

File: packages/components/src/button/src/Button.tsx (hunk 1 above)
Comment (fraincs) [reply packages/components/src/button/src/Button.tsx:147]: It has been decided that the current font size on small buttons is too small. md is the minimum a button text should be only the height changes between a small and medium button now.

File: packages/components/src/tag/src/Tag.tsx (hunk 2 above)
Comment (fraincs) [reply packages/components/src/tag/src/Tag.tsx:84]: It has been decided that the current font size on small tags is too small. md is the minimum a tag text should be only the height changes between a small and medium tag now.

**Comment by fraincs** [comment]
> 309 chromatic changes 😱 I hope you won’t miss anything!

This is what happens when you change one thing on a button. Buttons are everywhere.

File: packages/components/src/button/src/Button.tsx (hunk 1 above)
Comment (patricklafrance) [reply packages/components/src/button/src/Button.tsx:147]: Got it but it should still allow a responsive syntax.

Comment (fraincs) [reply packages/components/src/button/src/Button.tsx:147]: enlighten me on the how to do it :)

**Comment by Chicoalvi** [comment]
> Should a Button and a Tag have the same border color?
> 
> image

It's just my POV.

The whole area of a button is cleavable ≠ The clickable area (for closing) of the tag is just the area of the cross button
>so the graphic rendering of hovering is different

As user, i easy identify the differents components side by side. I know that are different and that they haven’t the same function

File: packages/components/src/button/src/Button.tsx (hunk 1 above)
Comment (patricklafrance) [reply packages/components/src/button/src/Button.tsx:147]: > > Should a Button and a Tag have the same border color?
> > image
> 
> It's just my POV.
> 
> The whole area of a button is cleavable ≠ The clickable area (for closing) of the tag is just the area of the cross button
> 
> > so the graphic rendering of hovering is different
> 
> As user, i easy identify the differents components side by side. I know that are different and that they haven’t the same function

@Chicoalvi make sense, thanks!

Comment (patricklafrance) [reply packages/components/src/button/src/Button.tsx:147]: > enlighten me on the how to do it :)

You can do it with `useResponsiveValue`. To enforce the the minimum size constraint, you either use either `createSizeAdapter` or `adaptSize`.

DM me if you need help.

**Comment by fraincs** [comment]
> As user, i easy identify the differents components side by side. I know that are different and that they haven’t the same function

I think the difference is not subtle enough. Thoughts?

**Comment by fraincs** [comment]
It has been discussed that a TagList should be solid by default.

image

**Review by patricklafrance — APPROVED** [review]

**Pull request merged**